File: problems/2299-strong-password-checker-ii/solution.cpp
class Solution {
public:
    bool strongPasswordCheckerII(string password) {
        if (password.length() < 8) return false;
        bool hasLower = false, hasUpper = false, hasDigit = false, hasSpecial = false;
        string specialChars = "!@#$%^&*()-+";
        for (char c : password) {
            if (islower(c)) hasLower = true;
            if (isupper(c)) hasUpper = true;
            if (isdigit(c)) hasDigit = true;
            if (specialChars.find(c) != string::npos) hasSpecial = true;
        }
        if (!hasLower || !hasUpper || !hasDigit || !hasSpecial) return false;
        for (int i = 0; i < password.length() - 1; i++) {
            if (password[i] == password[i + 1]) return false;
        }
        return true;
    }
};

File: problems/2299-strong-password-checker-ii/solution.py
class Solution:
    def strongPasswordCheckerII(self, password: str) -> bool:
        if len(password) < 8:
            return False
        lower = False
        upper = False
        digit = False
        special = False
        special_chars = "!@#$%^&*()-+"
        for i in range(len(password)):
            if password[i].islower():
                lower = True
            elif password[i].isupper():
                upper = True
            elif password[i].isdigit():
                digit = True
            elif password[i] in special_chars:
                special = True
            if i > 0 and password[i] == password[i - 1]:
                return False
        return lower and upper and digit and special